massa-execution-worker/src/context.rs

```diff
@@ -908,6 +908,13 @@ impl ExecutionContext {
             }
         }
 
+        // take the ledger changes again to take into account the balance change of canceled messages
+        let ledger_changes_canceled = self.speculative_ledger.take();
+
+        for (address, change) in ledger_changes_canceled.0 {
```

Comment:
haven't do rust for a while but this should works : 
```rust 
ledger_changes.0.extend(ledger_changes_canceled.0);
```

merge two collection

Reply:
It seems that we should use ledger_changes.apply(...) instead!

Reply:
This is outdated, resolving as a result